apps/web/pages/post/[id].tsx

```diff
@@ -1,16 +1,41 @@
-const Post = () => {
+import { useRouter } from "next/router";
+import { gql, useQuery } from "@apollo/client";
+import Spinner from "../../components/Spinner";
+import { GetPostByIdResult } from "../../interface/post"
+
+const GET_POST = gql`
+  query Query($page_id: ID!) {
+  getPostById(id: $page_id) {
+    content
+    id
+  }
+}
+`;
+
+const PostDesc = () => {
+  const router = useRouter();
```

Comment:
[nit] We can destructure just what we need here:
```suggestion
  const { query, push } = useRouter();
```